check/windows/winbase.lua

```diff
@@ -163,4 +167,31 @@ function WindowsPowershellCmdletCheck:run(callback)
   end
 end
 
+function convertTableToString(obj)
+    if type(obj) == "table" then
+        local stringVal=""
+        local isFirst = true
+        for index, data in pairs(obj) do
+                if index == "message" then
+                    stringVal = data
+                    break;
+                end
+                if isFirst then 
+                        isFirst = false
+                else    
+                        stringVal = stringVal .. ", "
+                end
+                stringVal = stringVal .. index .. " = "
+                if type(data) == "table" then
+                        stringVal = stringVal .. convertTableToString(data)
+                else    
+                        stringVal = stringVal .. data
+                end
+        end
+        return "{" .. stringVal .. "}"
+    else
+        return obj
+    end  
+end
```

Comment:
This is a non-cohesive and overly-complex function. It doesn't convert a table to a string, as the name implies. Instead, it tries to do all the work of converting an error value into an appropriate error message. This should be at least two functions: one that converts error values to messages and one that turns entire tables into strings.

This function will also produce a bad value for the case where a table contains a "message". For a table like `{message="an error happened"}`, it'll return `"{an error happened}"`. It would be worth attempting to write a few tests for this new code to ensure it behaves appropriately.